### PR TITLE
fix: fix transactionList

### PR DESCRIPTION
**Pull request by LizavetaY**
### Description

Add transactionList to add pages with transactions

### Type of change

- [ ] Feature (introduces new functionality)
- [x] Bugfix
- [x] Configuration
- [ ] Refactoring (doesn't change functionality, only the code)
- [ ] Style changes (doesn't affect functionality, only the look of the components)

### PR Checklist

- [x] Synced up with latest from the main branch/Code is linted
- [x] Description of the changes has been added (Screenshots/Video)
- [x] Chakra components/constants/hooks have been used wherever possible

> If not explain here

- [x] Commit messages follow [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#summary)
- [x] Console has been checked for warnings

#### Screenshots/Video before change

Add screenshots from the [current deployed state](https://sandbox-money-app.herokuapp.com/) of the changed components here.

#### Screenshots/Video after change

https://user-images.githubusercontent.com/47990337/184409746-b67a9e85-3bf4-41d1-873e-5a1b13e3257b.mov

https://user-images.githubusercontent.com/47990337/184409778-8169c885-3977-422b-94bd-9935ba6d60e0.mov

https://user-images.githubusercontent.com/47990337/184409784-1a2341af-4f69-40df-a9f5-6dea674327d8.mov

### CODE REVIEW

File: src/components/TransactionItem/TransactionItem.jsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  }
+
+  return (
+    <>
+      {!!dataCategories &&
+        !!dataCategories.data &&
+        isFetchedCategories &&
+        !!dataWallets &&
+        !!dataWallets.data &&
+        isFetchedWallets && (
```
Comment (itsnotpolza) [inline src/components/TransactionItem/TransactionItem.jsx:60]: can you move this to a separate variable? it's quite unreadable

File: src/components/TransactionItem/TransactionItem.jsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+export const TransactionItem = ({
+  transactionData,
+  onEdit,
+  onDelete,
+  isExpensesType,
+  isShortView
+}) => {
+  let categoryData = {};
+  let subcategoryName = '';
+  let walletData = {};
```
Comment (itsnotpolza) [inline src/components/TransactionItem/TransactionItem.jsx:21]: i think this should be a state

File: src/pages/WalletView.jsx [hunk 3]
```diff
@@ -61,104 +52,85 @@ export const WalletView = () => {
...
+    !!dataExpenses &&
+    !!dataWallets &&
+    !!dataIncomes.data &&
+    !!dataExpenses.data &&
+    !!dataIncomes.data.incomes &&
+    !!dataExpenses.data.expenses &&
+    !!dataWallets.data &&
+    isFetchedIncomes &&
+    isFetchedExpenses &&
+    isFetchedWallets
```
Comment (itsnotpolza) [inline src/pages/WalletView.jsx:79]: this is getting out of hand, you've got to refactor this

File: src/pages/Landing.jsx [hunk 4]
```diff
@@ -63,12 +63,15 @@ export const Landing = () => {
...
-      .forEach((transaction) => {
+      .map((transaction) => {
         let wallet = dataWallets.data.find(
           (wallet) => wallet.id === transaction.walletId
         );
+
         transaction.currency = wallet.currency;
       });
+
+    allTransactions = allTransactions.slice(0, 10);
```
Comment (kalashnikau) [inline src/pages/Landing.jsx:74]: does it works?

File: src/components/TransactionItem/TransactionItem.jsx (hunk 1 above)
Comment (LizavetaY) [reply src/components/TransactionItem/TransactionItem.jsx:60]: Снимок экрана 2022-08-12 в 9 23 51 PM
Снимок экрана 2022-08-12 в 9 24 00 PM

File: src/pages/WalletView.jsx (hunk 3 above)
Comment (LizavetaY) [reply src/pages/WalletView.jsx:79]: Снимок экрана 2022-08-12 в 9 26 01 PM
Снимок экрана 2022-08-12 в 9 26 09 PM

File: src/pages/Landing.jsx (hunk 4 above)
Comment (LizavetaY) [reply src/pages/Landing.jsx:74]: Снимок экрана 2022-08-12 в 9 27 33 PM

File: src/components/TransactionItem/TransactionItem.jsx (hunk 2 above)
Comment (LizavetaY) [reply src/components/TransactionItem/TransactionItem.jsx:21]: Снимок экрана 2022-08-12 в 9 25 02 PM

File: src/components/TransactionItem/TransactionItem.jsx [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    isFetched: isFetchedWallets
+  } = useQuery(['wallets'], getWallets);
+
+  const getCategory = () =>
+    (dataCategories || []).find(
+      (category) => category.id === transactionData.categoryId
+    );
+
+  const getSubcategoryName = () => {
+    const category = getCategory();
```
Comment (kalashnikau) [inline src/components/TransactionItem/TransactionItem.jsx:37]: what if here is undefined?

File: src/components/TransactionItem/TransactionItem.jsx [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+            >
+              <Box mr="10px" minW="85px" w="15%" textAlign="center">
+                {format(
+                  new Date(transactionData.dateOfTransaction),
+                  'dd.MM.yyyy'
+                )}
+              </Box>
+
+              <Box mr="10px" w="25%" textAlign="center">
+                <Text>{getCategory().name}</Text>
```
Comment (kalashnikau) [inline src/components/TransactionItem/TransactionItem.jsx:83]: can be undefined, so possible js error

File: src/components/TransactionItem/TransactionItem.jsx [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+                {isExpensesType && !!getSubcategoryName().length && (
+                  <Text>{() => getSubcategoryName()}</Text>
```
Comment (kalashnikau) [inline src/components/TransactionItem/TransactionItem.jsx:85]:
```suggestion
                  <Text>{getSubcategoryName()}</Text>
```

File: src/components/TransactionItem/TransactionItem.jsx [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+              <Text mr="10px" w="20%" textAlign="center">
+                {`${isExpensesType ? '-' : '+'} ${transactionData.value} ${
+                  transactionData.currency.symbol
+                }`}
+              </Text>
+
+              <Box mr="10px" w="15%" textAlign="center">
+                <Tooltip w="100%" label={getWallet().name}>
+                  <Text cursor="pointer">
+                    Wallet {transactionData.currency.currencyCode}
```
Comment (kalashnikau) [inline src/components/TransactionItem/TransactionItem.jsx:98]: need to be localized i18n

File: src/components/TransactionItem/TransactionItem.jsx [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+              </Box>
+
+              <Text mr="10px" w="20%" textAlign="center">
+                {`${isExpensesType ? '-' : '+'} ${transactionData.value} ${
+                  transactionData.currency.symbol
+                }`}
+              </Text>
+
+              <Box mr="10px" w="15%" textAlign="center">
+                <Tooltip w="100%" label={getWallet().name}>
```
Comment (kalashnikau) [inline src/components/TransactionItem/TransactionItem.jsx:96]: also can be undefined

File: src/components/TransactionList/TransactionList.jsx [hunk 10]
```diff
@@ -1,115 +1,207 @@
...
+      )}
+      {!!Object.keys(chosenTransactionData).length && (
+        <ConfirmationModal
+          isOpen={deleteTransactionModal.isOpen}
+          onSubmit={deleteOnSubmit}
+          onClose={deleteTransactionModal.onClose}
+          title={
+            chosenTransactionData.transactionType === 'Income'
+              ? i18next.t('modal.deleteIncome.title')
+              : i18next.t('modal.deleteExpense.title')
```
Comment (kalashnikau) [inline src/components/TransactionList/TransactionList.jsx:196]:
```suggestion
           i18next.t(`modal.delete${chosenTransactionData.transactionType}.title`)
```

File: src/pages/Expenses.jsx [hunk 11]
```diff
@@ -111,44 +51,21 @@ export const Expenses = () => {
...
-        />
+      {isFetchedExpenses && !allTransactions.length ? (
+        <Text color={textColor} fontSize="xl">
+          {i18next.t('transaction.noData')}
+        </Text>
+      ) : (
+        <TransactionList list={allTransactions} maxH="570px" isExpensesType />
       )}
+
+      {!isFetchedExpenses ? <Preloader /> : null}
```
Comment (kalashnikau) [inline src/pages/Expenses.jsx:68]:
```suggestion
      {!isFetchedExpenses && <Preloader />}
```

File: src/pages/Incomes.jsx [hunk 12]
```diff
@@ -1,88 +1,55 @@
...
+    >
+      {isFetchedIncomes && !allTransactions.length ? (
+        <Text color={textColor} fontSize="xl">
+          {i18next.t('transaction.noData')}
+        </Text>
+      ) : (
+        <TransactionList list={allTransactions} maxH="570px" />
+      )}
+
+      {!isFetchedIncomes ? <Preloader /> : null}
```
Comment (kalashnikau) [inline src/pages/Incomes.jsx:52]:
```suggestion
      {!isFetchedIncomes && <Preloader />}
```

File: src/pages/WalletView.jsx [hunk 13]
```diff
@@ -61,108 +50,76 @@ export const WalletView = () => {
...
+
+  if (
+    !!dataIncomes &&
+    !!dataExpenses &&
+    !!dataWallets &&
+    isFetchedIncomes &&
+    isFetchedExpenses &&
+    isFetchedWallets
+  ) {
+    allTransactions = [...dataIncomes, ...dataExpenses]
```
Comment (kalashnikau) [inline src/pages/WalletView.jsx:74]: i saw the same code a few times, so maybe let's move the function to some helper?

File: src/components/TransactionItem/TransactionItem.jsx (hunk 5 above)
Comment (LizavetaY) [reply src/components/TransactionItem/TransactionItem.jsx:37]: Снимок экрана 2022-08-12 в 9 53 13 PM

File: src/components/TransactionItem/TransactionItem.jsx (hunk 6 above)
Comment (LizavetaY) [reply src/components/TransactionItem/TransactionItem.jsx:83]: Снимок экрана 2022-08-12 в 9 54 11 PM

Comment (LizavetaY) [reply src/components/TransactionItem/TransactionItem.jsx:83]: Снимок экрана 2022-08-12 в 9 56 16 PM

File: src/components/TransactionItem/TransactionItem.jsx (hunk 7 above)
Comment (LizavetaY) [reply src/components/TransactionItem/TransactionItem.jsx:85]: Снимок экрана 2022-08-12 в 9 57 36 PM

File: src/components/TransactionItem/TransactionItem.jsx (hunk 8 above)
Comment (LizavetaY) [reply src/components/TransactionItem/TransactionItem.jsx:98]: Снимок экрана 2022-08-12 в 9 59 49 PM

File: src/components/TransactionItem/TransactionItem.jsx (hunk 9 above)
Comment (LizavetaY) [reply src/components/TransactionItem/TransactionItem.jsx:96]: Снимок экрана 2022-08-12 в 10 00 20 PM

File: src/components/TransactionList/TransactionList.jsx (hunk 10 above)
Comment (LizavetaY) [reply src/components/TransactionList/TransactionList.jsx:196]: Снимок экрана 2022-08-12 в 10 03 17 PM

File: src/pages/Expenses.jsx (hunk 11 above)
Comment (LizavetaY) [reply src/pages/Expenses.jsx:68]: Снимок экрана 2022-08-12 в 10 05 31 PM

File: src/pages/Incomes.jsx (hunk 12 above)
Comment (LizavetaY) [reply src/pages/Incomes.jsx:52]: Снимок экрана 2022-08-12 в 10 05 53 PM

File: src/pages/WalletView.jsx (hunk 13 above)
Comment (LizavetaY) [reply src/pages/WalletView.jsx:74]: Снимок экрана 2022-08-12 в 10 28 39 PM

**Review by kalashnikau — APPROVED** [review]

**Review by PavelTitov16 — APPROVED** [review]

**Pull request merged**